packages/design-system/src/components/IdleTimeout/IdleTimeout.tsx

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable react/no-multi-comp */
+/* eslint-disable react/no-danger */
+import React from 'react';
+import { Dialog } from '../Dialog';
+import { Button } from '../Button';
+
+export interface IdleTimeoutProps {
+  /**
+   * The text for the 'close' button on the warning dialog
+   */
+  closeDialogText?: string;
+  /**
+   * The text for the 'continue session' button in warning dialog.
+   */
+  continueSessionText?: string;
+  /**
+   * The heading text for the warning dialog.
+   */
+  heading?: string;
+  /**
+   * The text for the button that ends the session in warning dialog.
+   */
+  endSessionButtonText?: string;
+  /**
+   * The message text for the warning dialog.
+   * Note that using the token `<timeToTimeout>` will be replaced in the message text with the number of minutes until timeout.
+   */
+  message?: string;
```

Comment:
It could possibly be a function that takes a number and returns a string `message?: (t : number) => string;` with that number available to interpolate then down below it could be `div>{{message(timeToTimeout)}}</div>` which might be a nice way to present it and would allow the user to change that time to minutes:seconds, or just seconds or whatever they wanted..